src/main/java/forestry/plugins/compat/PluginNatura.java

```diff
@@ -119,145 +167,55 @@ public void registerCrates() {
-		}
-		if (logNatura != null) {
-			crateRegistry.registerCrate(new ItemStack(logNatura, 1, 0));
-			crateRegistry.registerCrate(new ItemStack(logNatura, 1, 1));
-			crateRegistry.registerCrate(new ItemStack(logNatura, 1, 2));
-			crateRegistry.registerCrate(new ItemStack(logNatura, 1, 3));
-			crateRegistry.registerCrate(new ItemStack(logWillow, 1, 0));
-		}
-		Item bloodWood = GameRegistry.findItem(NATURA, "bloodwood");
-		if (bloodWood != null) {
-			crateRegistry.registerCrate(new ItemStack(bloodWood, 1, 0));
-		}
-		Item darkTree = GameRegistry.findItem(NATURA, "Dark Tree");
-		if (darkTree != null) {
-			crateRegistry.registerCrate(new ItemStack(darkTree, 1, 0));
-			crateRegistry.registerCrate(new ItemStack(darkTree, 1, 1));
-		}
-		Item heatSand = GameRegistry.findItem(NATURA, "heatsand");
-		if (heatSand != null) {
-			crateRegistry.registerCrate(new ItemStack(heatSand, 1, 0));
-		}
-		Item taintedSoil = GameRegistry.findItem(NATURA, "soil.tainted");
-		if (taintedSoil != null) {
-			crateRegistry.registerCrate(new ItemStack(taintedSoil, 1, 0));
-		}
+		Iterables.concat(
+				edibles, logs, saplings, shrooms,
+				materials, seeds, berries, fruits,
+				crops
+		).forEach(item -> crateRegistry.registerCrate(item));
 	}
 
 	@Override
 	public void registerRecipes() {
-		Item seed = GameRegistry.findItem(NATURA, "barley.seed");
-		ItemStack seedBarley = new ItemStack(seed, 1, 0);
-		ItemStack seedCotton = new ItemStack(seed, 1, 1);
-
-		ArrayList<ItemStack> seedList = new ArrayList<>();
-		if (seedBarley != null) {
-			seedList.add(seedBarley);
-			RecipeManagers.moistenerManager.addRecipe(seedBarley, new ItemStack(Blocks.MYCELIUM), 5000);
-		}
-		if (seedCotton != null) {
-			seedList.add(seedCotton);
-		}
-
 		int amount = ForestryAPI.activeMode.getIntegerSetting("squeezer.liquid.seed");
-		for (ItemStack aSeedList : seedList) {
-			RecipeManagers.squeezerManager.addRecipe(10, new ItemStack[]{aSeedList}, Fluids.SEED_OIL.getFluid(amount));
+		for (ItemStack seed : seeds) {
+			RecipeManagers.squeezerManager.addRecipe(10, new ItemStack[]{seed}, Fluids.SEED_OIL.getFluid(amount));
+			RecipeManagers.moistenerManager.addRecipe(seed, new ItemStack(Blocks.MYCELIUM), 5000);
 		}
 
 		if (ForestryAPI.enabledPlugins.contains(ForestryPluginUids.FARMING)) {
-			Block cropBlock = GameRegistry.findBlock(NATURA, "N Crops");
-			//TODO: Natura for 1.9
-			//			Farmables.farmables.get("farmWheat").add(new FarmableHandPlanted(seedBarley, cropBlock, 3));
-			//			Farmables.farmables.get("farmWheat").add(new FarmableHandPlanted(seedCotton, cropBlock, 8));
+			cropBlocks.forEach(itemStack -> {
+				Block block = ItemStackUtil.getBlock(itemStack);
+				ItemStack seedItem;
+				int maxAge;
+				try {
+					// XXX: getMaxAge
+					maxAge = (int)block.getClass().getDeclaredMethod("getMaxAge").invoke(block);
+
+					// XXX: find seed items
+					seedItem = block.getPickBlock(block.getBlockState().getBaseState(), null, null, null, null);
```

Comment:
This section is suuuper hacky.
I'd rather list out properties like (blockName, itemName, maxAge) for each crop instead of trying to get their properties automatically.